modules/entity/src/sorted_state_adapter.ts

```diff
@@ -72,8 +73,16 @@ export function createSortedStateAdapter<T>(selectId: any, sort: any): any {
   }
 
   function updateManyMutably(updates: Update<T>[], state: R): DidMutate;
-  function updateManyMutably(updates: any[], state: any): DidMutate {
+  function updateManyMutably(map: Map<T>, state: R): DidMutate;
+  function updateManyMutably(updatesOrMap: any[] | any, state: any): DidMutate {
     const models: T[] = [];
+    const updates: Update<T>[] =
+      updatesOrMap instanceof Array
+        ? updatesOrMap
+        : state.ids.map((id: string | number) => ({
```

Comment:
I think that creating an `Update` object for each entity means even entities that shouldn't change will lose referential equality. I think this `map` should be a `reduce` instead that shrinks the `ids` array into a smaller array of `Update`s for only changed entities.